build.sbt

```diff
@@ -14,37 +15,74 @@
  */
 
 import Dependencies._
+import SparkVersionAxis._
+import JacocoSetup._
 
-ThisBuild / organization := "za.co.absa"
+ThisBuild / organization := "za.co.absa.atum-service"
 
 lazy val scala211 = "2.11.12"
 lazy val scala212 = "2.12.12"
+lazy val spark2 = "2.4.7"
+lazy val spark3 = "3.3.1"
+
+ThisBuild / crossScalaVersions := Seq(scala211, scala212)
+ThisBuild / scalaVersion := scala212
 
-Test / parallelExecution := false
+ThisBuild / versionScheme := Some("early-semver")
 
 Global / onChangedBuildSource := ReloadOnSourceChanges
 
-lazy val root = (project in file("."))
+lazy val printScalaVersion = taskKey[Unit]("Print Scala versions for atum-service is being built for.")
+
+ThisBuild / printScalaVersion := {
+  val log = streams.value.log
+  log.info(s"Building with Scala ${scalaVersion.value}")
+}
+
+lazy val commonSettings = Seq(
+  libraryDependencies ++= commonDependencies,
+  scalacOptions ++= Seq("-unchecked", "-deprecation", "-feature", "-Xfatal-warnings"),
+  javacOptions ++= Seq("-source", "1.8", "-target", "1.8", "-Xlint"),
+  Test / parallelExecution := false
+)
+
+lazy val parent = (project in file("."))
+  .aggregate(atumServer.projectRefs ++ atumAgent.projectRefs: _*)
+  .settings(
+    name := "atum-service-parent",
+    publish / skip := true
+  )
+
+lazy val atumAgent = (projectMatrix in file("agent"))
   .settings(
-    name := "atum-root",
-    javacOptions ++= Seq("-source", "1.8", "-target", "1.8", "-Xlint")
+    commonSettings ++ Seq(
+      name := "atum-agent",
+      (Compile / compile) := ((Compile / compile) dependsOn printScalaVersion).value,
+      scalafmtOnCompile := true
+    )
   )
+  .enablePlugins(ScalafmtPlugin)
+  .sparkRow(SparkVersionAxis(spark2), scalaVersions = Seq(scala211, scala212))
+  .sparkRow(SparkVersionAxis(spark3), scalaVersions = Seq(scala212))
 
-lazy val server = project
+lazy val atumServer = (projectMatrix in file("server"))
+  .settings(
+    commonSettings ++ Seq(
+      name := "atum-server",
+      libraryDependencies ++= Dependencies.serverDependencies,
+      (Compile / compile) := ((Compile / compile) dependsOn printScalaVersion).value,
+      webappWebInfClasses := true,
+      inheritJarManifest := true
+    ): _*
+  )
   .settings(
-    name         := "atum-server",
-    scalaVersion := scala212,
-    libraryDependencies ++= Dependencies.serverDependencies,
-    webappWebInfClasses := true,
-    inheritJarManifest  := true
+    jacocoReportSettings := jacocoSettings( scalaVersion.value, "atum-server"),
+    jacocoExcludes := jacocoProjectExcludes()
   )
   .enablePlugins(TomcatPlugin)
   .enablePlugins(AutomateHeaderPlugin)
+  .jvmPlatform(scalaVersions = Seq(scala212))
```

Comment:
won't scala 2.11 be supported?

Reply:
It will be supported but again, this is a new library.

Reply:
Good for now. 2.13 should be added eventually too.